### ISSUE TITLE
react-native-git-upgrade doesn't work with NPM 5

### ISSUE THREAD

**Issue by Strayer** [facebook/react-native#14239]
### Description

Tried upgrading to react-native 0.44.2 with react-native-git-upgrade@0.2.7 and npm@5.0.0 and received the following error:

```
$ react-native-git-upgrade
git-upgrade info Check for updates
git-upgrade info Using yarn 0.24.6
git-upgrade info Read package.json files
git-upgrade info Check declared version
git-upgrade info Check matching versions
git-upgrade info Check React peer dependency
git-upgrade info Check that Git is installed
git-upgrade info Get information from NPM registry
git-upgrade ERR! An error occurred during upgrade:
git-upgrade ERR! Error: The specified version of React Native undefined doesn't exist.
Re-run the react-native-git-upgrade command with an existing version,
for example: "react-native-git-upgrade 0.38.0",
or without arguments to upgrade to the latest: "react-native-git-upgrade".
    at parseInformationJsonOutput (/Users/strayer/.nvm/versions/node/v7.10.0/lib/node_modules/react-native-git-upgrade/cliEntry.js:110:11)
    at Object.<anonymous> (/Users/strayer/.nvm/versions/node/v7.10.0/lib/node_modules/react-native-git-upgrade/cliEntry.js:264:48)
    at Generator.next (<anonymous>)
    at step (/Users/strayer/.nvm/versions/node/v7.10.0/lib/node_modules/react-native-git-upgrade/cliEntry.js:171:191)
    at /Users/strayer/.nvm/versions/node/v7.10.0/lib/node_modules/react-native-git-upgrade/cliEntry.js:171:361
    at tryCallOne (/Users/strayer/.nvm/versions/node/v7.10.0/lib/node_modules/react-native-git-upgrade/node_modules/promise/lib/core.js:37:12)
    at /Users/strayer/.nvm/versions/node/v7.10.0/lib/node_modules/react-native-git-upgrade/node_modules/promise/lib/core.js:123:15
    at flush (/Users/strayer/.nvm/versions/node/v7.10.0/lib/node_modules/react-native-git-upgrade/node_modules/asap/raw.js:50:29)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

It works fine after downgrading to npm@4.6.1

### Reproduction Steps and Sample Code

Run react-native-git-upgrade in an environment that has npm 5 in its path.

### Solution

It seems like something goes wrong in [cliEntry.js in parseInformationJsonOutput](https://github.com/facebook/react-native/blob/master/react-native-git-upgrade/cliEntry.js#L99). The output of npm view react-native@latest --json seems equal, but I may be mistaken.

### Additional Information

* React Native version: 0.44.0
* Development Operating System: macOS Sierra 10.12.5
* NodeJS v7.10.0, yarn version v0.24.6

**Comment by ncuillery** [comment]
Thanks for reporting. I've added some logs in the `exec` function:
https://github.com/ncuillery/react-native/commit/6dff9749af9384e8d7df9bb93dbbfe834c32d1af

#### NPM 5.0.0
```
git-upgrade info Get information from NPM registry 
COMMAND npm view react-native@latest --json
DATA 8192
EXIT 0
git-upgrade ERR! An error occurred during upgrade: 
git-upgrade ERR! Error: The specified version of React Native undefined doesn't exist.
```

#### NPM 4.6.1
```
git-upgrade info Get information from NPM registry 
COMMAND npm view react-native@latest --json
DATA 8192
DATA 8192
DATA 4401
EXIT 0
git-upgrade info Upgrading to React Native 0.44.2, React 16.0.0-alpha.6 
```

It seems that the child process spawned to grab RN package information from NPM is exited right after the reception of the first chunk of data.

The process is spawned by [shelljs](https://github.com/shelljs/shelljs#execcommand--options--callback).

Still investigating.

**Comment by ncuillery** [comment]
I reproduced the problem by only using Node APIs:
https://github.com/npm/npm/issues/16780

**Comment by ncuillery** [comment]
Fix by the (awesome) npm team in NPM@5.0.1

@facebook-github-bot close
